### PR TITLE
update for lean 3.4.2

### PR DESCRIPTION
**Pull request by pnmadelaine**

### CODE REVIEW

**Comment by robertylewis** [comment]
Nice!

**Comment by pnmadelaine** [comment]
I'm note done yet, I just managed to remove the build errors before it gets to the tests.
Plus there still are some 'exit' commands in the code.

File: leanpkg.toml [hunk 1]
```diff
@@ -1,8 +1,9 @@
 [package]
 name = "polya"
 version = "0.1"
-lean_version = "3.3.0"
+lean_version = "3.4.2"
+path = "src"
 
 [dependencies]
-mathlib = {git = "https://github.com/leanprover/mathlib", rev = "lean-3.3.0"}
+mathlib = {path = "/home/pnm/leanprover/mathlib/"}
```
Comment (robertylewis) [inline leanpkg.toml]: Can you remove this change? The version update is good, and you can keep this local. (Maybe add `leanpkg.toml` to .gitignore.)

File: src/interactive.lean [hunk 2]
```diff
@@ -71,47 +59,60 @@ namespace interactive
...
+    put ⟨ps.sum_cache, ps.prod_cache, bb'⟩
 
 meta def add_hypotheses (ns : parse (many ident)) : polya_tactic unit :=
-λ ps,
-do exps ← ns.mmap tactic.get_local,
-   bb' ← add_proofs_to_blackboard ps.bb exps,
-   return ((), ⟨ps.sum_cache, ps.prod_cache, bb'⟩)
+do 
+    ps ← get,
+    exps ← ns.mmap ↑tactic.get_local, -- TODO: is it different than ↑(ns.mmap tactic.get_local) ?
```
Comment (robertylewis) [inline src/interactive.lean]: Shouldn't matter.

File: src/proof_trace.lean [hunk 3]
```diff
@@ -83,10 +83,17 @@ private meta def ineq_proof.sketch_aux (ss : Π {e c}, sign_proof e c → tactic
...
-private meta def sign_proof.sketch_aux (is : Π {lhs rhs iq}, ineq_proof lhs rhs iq → tactic proof_sketch) (es : Π {lhs rhs c}, eq_proof lhs rhs c → tactic proof_sketch)  (sfs : Π {sfc}, sum_form_proof sfc → tactic proof_sketch) : Π {e c}, sign_proof e c → tactic proof_sketch
+private meta def sign_proof.sketch_aux
+     (is : Π {lhs rhs iq}, ineq_proof lhs rhs iq → tactic proof_sketch)
+     (es : Π {lhs rhs c}, eq_proof lhs rhs c → tactic proof_sketch)
+     (sfs : Π {sfc}, sum_form_proof sfc → tactic proof_sketch) :
+     Π {e c}, sign_proof e c → tactic proof_sketch
 | _ _ (sign_proof.hyp e c _) := 
   do s ← format_sign e c,
      return ⟨s, "hypothesis", []⟩
+| _ _ (sign_proof.scaled_hyp e c _ _) :=
```
Comment (robertylewis) [inline src/proof_trace.lean:94]: Was this case just missing before?

**Pull request closed without merging**